rest/route/distro.go

```diff
@@ -18,6 +18,47 @@ import (
 	"github.com/pkg/errors"
 )
 
+///////////////////////////////////////////////////////////////////////
+//
+// DELETE /rest/v2/distros/{distro_id}
+
+type distroIDDeleteHandler struct {
+	distroId string
+	sc       data.Connector
+}
+
+func makeDeleteDistroByID(sc data.Connector) gimlet.RouteHandler {
+	return &distroIDDeleteHandler{
+		sc: sc,
+	}
+}
+
+func (h *distroIDDeleteHandler) Factory() gimlet.RouteHandler {
+	return &distroIDDeleteHandler{
+		sc: h.sc,
+	}
+}
+
+// Parse() fetches the distroId from the http request.
+func (h *distroIDDeleteHandler) Parse(ctx context.Context, r *http.Request) error {
+	h.distroId = gimlet.GetVars(r)["distro_id"]
+	return nil
+}
+
+func (h *distroIDDeleteHandler) Run(ctx context.Context) gimlet.Responder {
+	_, err := h.sc.FindDistroById(h.distroId)
```

Comment:
It seems slightly odd to find the distro and then try deleting it. Would just attempting to delete it have the same effect, or is this useful validation?